teek/_widgets/treeview.py

```diff
@@ -0,0 +1,545 @@
+
+        for name, value in self._creation_opts.items():
+            self.config[name] = value
+
+        if self._text is not None:
+            self.heading.config['text'] = self._text
+
+    def to_tcl(self):
+        # Only name in column list
+        return self._name
+
+
+class TreeviewRowList(MutableSequence):
+    """
+    List containing all rows of a treeview. This class is used for all rows and
+    nested rows (subrows).
+
+    Note, that instances of :class:`TreeviewRowList` must not be created
+    manually. It is accessible through the `.rows` and `.subrows` attributes.
+    """
+    def __init__(self, treeview, root=None):
+        super().__init__()
+
+        self._root = root
+        self._treeview = treeview
+
+    def __repr__(self):
+        return '%s(root=%r, rows=%d)' % (
+            self.__class__.__name__,
+            self._root,
+            len(self)
+        )
+
+    @make_thread_safe
+    def __getitem__(self, index):
+        return self._get_all_rows()[index]
+
+    @make_thread_safe
+    def __setitem__(self, index, row):
+        del self[index]
+        self.insert(index, row)
+
+    @make_thread_safe
+    def __delitem__(self, index):
+        self._treeview._call(None, self._treeview, 'delete', self[index])
+
+    @make_thread_safe
+    def __len__(self):
+        return len(self._get_all_rows())
+
+    def _get_all_rows(self):
+        """
+        Get a list of all rows created for a root node.
+        """
+        ids = self._treeview._call([str], self._treeview, 'children',
+                                   self._root)
+        rows = []
+
+        for id_ in ids:
+            row = TreeviewRow(name=id_)
+            row._assign(treeview=self._treeview, root=self._root,
+                        set_config=False)
+            rows.append(row)
+
+        return rows
+
+    @make_thread_safe
+    def insert(self, index, row):
+        """
+        Insert a row or values converted to a row at the given index.
+        """
+        # Convert to row
+        if not isinstance(row, TreeviewRow):
+            row = TreeviewRow(values=row)
+
+        self._treeview._call(
+            None, self._treeview, 'insert', self._root, index, '-id', row
+        )
+
+        row._assign(treeview=self._treeview, root=self._root)
+
+    @make_thread_safe
+    def sort(self, column_pos, reverse=False):
+        """
+        Sort all rows according to value of column with given ``column_pos``.
+        The direction of the sorting can be set with ``reverse``. If this value
+        is true, the rows are sorted in ascending order.
+        """
+        if column_pos < 1:
+            raise IndexError('column position must be greater or equal 1')
+
+        sorted_rows = list(self).copy()
+        sorted_rows.sort(key=lambda r: r.config['values'][column_pos - 1],
+                         reverse=reverse)
+
+        for to_pos, row in enumerate(sorted_rows):
+            from_pos = self.index(row)
```

Comment:
This will be quite slow with a big treeview, because finding the index of a row is slow and you're doing it in a loop. (If you know some big-O notation, this code is `O(n^2)` while sorting is usually `O(n*log(n))`.) Maybe mention the possible slowness in the docstring?

Alternatively, you might be able to avoid the whole `self.index(row)` by just deleting the `if from_pos` check?